### PR TITLE
Really disable shuffling for gen_defaults tests

### PR DESCRIPTION
**Pull request by zanderso**
https://github.com/flutter/flutter/issues/142716

### CODE REVIEW

**Comment by vashworth** [comment]
@zanderso Does `flutter_test` need to be done too?

Seems there were also some errors in that part https://ci.chromium.org/ui/p/flutter/builders/prod/Mac_arm64%20framework_tests_misc/75/overview

```
 00:08 +154 ~1: /Volumes/Work/s/w/ir/x/w/flutter/packages/flutter_test/test/widget_tester_test.dart: group retry flag allows test to run multiple times the group with retry flag the test inside it    
 ══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
 The following TestFailure was thrown running a test:
 Expected: true
   Actual: <false>
 When the exception was thrown, this was the stack:
 #4      main.<anonymous closure>.<anonymous closure>.<anonymous closure> (file:///Volumes/Work/s/w/ir/x/w/flutter/packages/flutter_test/test/widget_tester_test.dart:56:9)
 #5      testWidgets.<anonymous closure>.<anonymous closure> (package:flutter_test/src/widget_tester.dart:183:29)
 <asynchronous suspension>
 #6      TestWidgetsFlutterBinding._runTestBody (package:flutter_test/src/binding.dart:1019:5)
 <asynchronous suspension>
 <asynchronous suspension>
 (elided one frame from package:stack_trace)
 This was caught by the test expectation on the following line:
   file:///Volumes/Work/s/w/ir/x/w/flutter/packages/flutter_test/test/widget_tester_test.dart line 56
 The test description was:
   the test inside it
 ════════════════════════════════════════════════════════════════════════════════════════════════════
 00:08 +154 ~1 -1: /Volumes/Work/s/w/ir/x/w/flutter/packages/flutter_test/test/widget_tester_test.dart: group retry flag allows test to run multiple times the group with retry flag the test inside it [E]
   Test failed. See exception logs above.
   The test description was: the test inside it
   
 Retry: group retry flag allows test to run multiple times the group with retry flag the test inside it
 ```
 
 ```
 00:09 +180 ~2: /Volumes/Work/s/w/ir/x/w/flutter/packages/flutter_test/test/widget_tester_test.dart: testWidget retry flag allows test to run multiple times the test with retry flag                   
 ══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
 The following TestFailure was thrown running a test:
 Expected: true
   Actual: <false>
 When the exception was thrown, this was the stack:
 #4      main.<anonymous closure>.<anonymous closure> (file:///Volumes/Work/s/w/ir/x/w/flutter/packages/flutter_test/test/widget_tester_test.dart:65:7)
 #5      testWidgets.<anonymous closure>.<anonymous closure> (package:flutter_test/src/widget_tester.dart:183:29)
 <asynchronous suspension>
 #6      TestWidgetsFlutterBinding._runTestBody (package:flutter_test/src/binding.dart:1019:5)
 <asynchronous suspension>
 <asynchronous suspension>
 (elided one frame from package:stack_trace)
 This was caught by the test expectation on the following line:
   file:///Volumes/Work/s/w/ir/x/w/flutter/packages/flutter_test/test/widget_tester_test.dart line 65
 The test description was:
   the test with retry flag
 ════════════════════════════════════════════════════════════════════════════════════════════════════
 00:09 +180 ~2 -1: /Volumes/Work/s/w/ir/x/w/flutter/packages/flutter_test/test/widget_tester_test.dart: testWidget retry flag allows test to run multiple times the test with retry flag [E]            
   Test failed. See exception logs above.
   The test description was: the test with retry flag
   
 Retry: testWidget retry flag allows test to run multiple times the test with retry flag
 ```

**Review by vashworth — APPROVED** [review]
LGTM with question above: https://github.com/flutter/flutter/pull/142721#issuecomment-1921790945

**Comment by zanderso** [comment]
@vashworth Thanks for spotting that. I've updated the PR to disable shuffling for gen_defaults tests (hopefully) the right way, and also added the tag to widget_tester_test.dart.

**Review by vashworth — APPROVED** [review]
Still LGTM!

File: packages/flutter_test/test/widget_tester_test.dart [hunk 1]
```diff
@@ -2,6 +2,12 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+// TODO(team-framework): Remove this tag once this test's state leaks/test
+// dependencies have been fixed.
+// https://github.com/flutter/flutter/issues/85160
+// Fails with "flutter test --test-randomize-ordering-seed=20240201"
+@Tags(<String>['no-shuffle'])
```
Comment (goderbauer) [inline packages/flutter_test/test/widget_tester_test.dart:9]: Please revert this. The test isn't failing. (Well, it fails ones and then is retried to verify the retry logic, what you saw in the log is the expected first failure. Overall the test passes with the seed.)

Comment (zanderso) [reply packages/flutter_test/test/widget_tester_test.dart]: Done.

**Pull request merged**